components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/SecurityHandlerAdapter.java

```diff
@@ -32,6 +44,27 @@
  */
 public abstract class SecurityHandlerAdapter implements InternalAPIHandler {
 
+    static HashMap<String, char[]> usersList;
+    static boolean useCarbonUserStore = false;
+    private static boolean isInitialized = false;
+
+    private static final Log LOG = LogFactory.getLog(SecurityHandlerAdapter.class);
+
+    public SecurityHandlerAdapter() throws CarbonException, XMLStreamException, IOException,
+            ManagementApiUndefinedException {
+        if (!isInitialized) {
+            ManagementApiParser mgtParser = new ManagementApiParser();
+            try {
+                usersList = mgtParser.getUserList();
+            } catch (UserStoreUndefinedException e) {
+                useCarbonUserStore = true;
+                LOG.info("User store config has not been defined in file " + ManagementApiParser.getConfigurationFilePath() + ". "
+                         + "Using carbon user store settings");
+            }
+            isInitialized = true;
```

Comment:
Shall we move this code block to a static method. Hence we can avoid updating static fields in the constructor.

Reply:
If we move it to a static method, we have no way of throwing an error while parsing. Also, we will not have a control over ordering if we move it to a static block. Therefore, I prefer leaving as it is. WDYT?

Reply:
done